crates/cxx-qt-gen/src/generator/cpp/fragment.rs

```diff
@@ -19,6 +19,19 @@ pub enum CppFragment {
     Source(String),
 }
 
+#[cfg(test)]
+impl Into<String> for CppFragment {
+    fn into(self) -> String {
+        match self {
+            CppFragment::Pair { header, source } => {
+                format!("Header:\n  {header}\nSource:\n  {source}")
+            }
+            CppFragment::Header(header) => header,
+            CppFragment::Source(source) => source,
+        }
+    }
+}
```

Comment:
this is not used i don't think? was it a previous attempt before the `require_header` / `require_pair` things?

Reply:
It's used in the require functions so I can format an error in the catchall block, e.g. if I'm matching:
```rust
CppFragment::Pair => { ... },
_ => Err(Error::new_spanned( ... )
```
I call fragment.clone.into() and it'll format that into the error. It lets me avoid matching Source, so requires less testing